### PR TITLE
Fix the implementation of CHECKCAST byte code instruction.

### PR DESCRIPTION
**Pull request by koutheir**
[`CHECKCAST`](https://docs.oracle.com/javase/specs/jvms/se11/html/jvms-6.html#jvms-6.5.checkcast) must not throw an exception when the object is `null`, even if the type cannot be resolved.

### CODE REVIEW

File: compiler/src/org.graalvm.compiler.java/src/org/graalvm/compiler/java/BytecodeParser.java [hunk 1]
```diff
@@ -1227,7 +1227,30 @@ protected void handleUnresolvedLoadConstant(JavaType type) {
      */
     protected void handleUnresolvedCheckCast(JavaType type, ValueNode object) {
         assert !graphBuilderConfig.unresolvedIsError();
-        append(new FixedGuardNode(graph.addOrUniqueWithInputs(IsNullNode.create(object)), Unresolved, InvalidateRecompile));
```
Comment (gilles-duboscq) [inline compiler/src/org.graalvm.compiler.java/src/org/graalvm/compiler/java/BytecodeParser.java:1230]: Was this change necessary? Or was it just for symmetry with the SVM code?

This structure will just end up being turned back into the `FixedGuardNode` so i think we should just keep this as is. Unless it fixes some issue i'm missing.

Comment (koutheir) [reply compiler/src/org.graalvm.compiler.java/src/org/graalvm/compiler/java/BytecodeParser.java:1230]: > Was this change necessary? Or was it just for symmetry with the SVM code?

Only for symmetry.

> This structure will just end up being turned back into the FixedGuardNode so i think we should just keep this as is. Unless it fixes some issue i'm missing.

I'll try that instead.

**Comment by mukel** [comment]
I'm integrating this PR, should be merged soon.
